x-pack/plugins/security_solution/public/common/utils/timeline/use_show_timeline.tsx

```diff
@@ -23,33 +24,36 @@ const DEPRECATED_HIDDEN_TIMELINE_ROUTES: readonly string[] = [
   '/manage',
 ];
 
-const isTimelineHidden = (currentPath: string, isGroupedNavigationEnabled: boolean): boolean => {
+const isTimelinePathVisible = (
+  currentPath: string,
+  isGroupedNavigationEnabled: boolean
+): boolean => {
   const groupLinksWithHiddenTimelinePaths = getLinksWithHiddenTimeline().map((l) => l.path);
 
   const hiddenTimelineRoutes = isGroupedNavigationEnabled
     ? groupLinksWithHiddenTimelinePaths
     : DEPRECATED_HIDDEN_TIMELINE_ROUTES;
 
-  return !!hiddenTimelineRoutes.find((route) => matchPath(currentPath, route));
+  return !hiddenTimelineRoutes.find((route) => matchPath(currentPath, route));
 };
 
 export const useShowTimeline = () => {
   const isGroupedNavigationEnabled = useIsGroupedNavigationEnabled();
   const { pathname } = useLocation();
   const { indicesExist, dataViewId } = useSourcererDataView(SourcererScopeName.timeline);
+  const userHasSecuritySolutionVisible = useKibana().services.application.capabilities.siem.show;
```

Comment:
The capabilities check has been moved from the security template_wrapper to this `useShowTimeline` hook, so all the timeline permissions logic is encapsulated here, preventing duplication.